### DISCUSSION TITLE
HID_INPUT_OUTPUT usage

### DISCUSSION THREAD

**Discussion by monurbstnc** [hathach/tinyusb#2080] [Q&A]
Hello I want to send data and receive data using belown macro.

#define TUD_HID_INOUT_DESCRIPTOR(_itfnum, _stridx, _boot_protocol, _report_desc_len, _epout, _epin, _epsize, _ep_interval) 

Here I did not understand the parameters of this macro. Anyone can help?

**Comment by hathach** [comment]
what you don't understand ? in any doubt, just check the code
